### ISSUE TITLE
Conflicts with internal Oauth2 app of Nextcloud12

### ISSUE THREAD

**Issue by poVoq** [soudis/oauth2#1]
Not sure if this has been replaced by the internal Oauth2 app in Nextcloud12, but since that one doesn't seem to implement real authentication features, maybe this app can still serve a purpose and could be altered to work without conflict in Nextcloud12?

**Comment by cigotete** [comment]
yes, I found in NextCloud12 that already exist an app in the apps directory with the name 'oauth2'. Seems that this soudis/oauth2 is to Nextcloud act as client, and the other is to Nextcloud act as Oauth provider.

**Comment by cigotete** [comment]
I have installed this soudis/oauth2 on a NextCloud 11 (fresh install), and I saw a form to add Oauth clients, so seems that is the same native feature that comes into Nextcloud 12 Admin Security section (or I am wrong?).

**Comment by poVoq** [comment]
Nextcloud12 does indeed have a similar thing integrated (apparently a complete rewrite), but it is totally undocumented and does not allow apps to access user data and thus can not be used as a sort of SSO.

**Comment by x9t9** [comment]
@poVoq  - I think it might be possible to get the user data from the API routers ( like [user provisioning](https://docs.nextcloud.com/server/12/admin_manual/configuration_user/user_provisioning_api.html) ?? ) .
Also - is this repository maintained ? Is it compatible with NC 12+ ?

**Comment by soudis** [comment]
this app is not maintained and not working, it was just a try (forked from owncloud/oauth2)

**Comment by x9t9** [comment]
@soudis - Ok. Thanks for the effort anyhow . I was meaning to do something similar myself. The OC code and architecture does not make for easy plugins writing ... :-)

**Comment by soudis** [comment]
true :) the best way would be to start from the rewritten app that's built in the nextcloud 12 core (https://github.com/nextcloud/server/tree/master/apps/oauth2) and extend it with an user provisioning end point. But I think I still lack the app writing knowledge to do that.
